src/Host.cpp

```diff
@@ -247,85 +313,40 @@ void Host::saveModules(int sync, bool backup)
             if (!packageDir.exists()) {
                 packageDir.mkpath(packagePathName);
             }
-        } else if (backup) {
-            savePath.rename(filename_xml, dirName + moduleName + time); //move the old file, use the key (module name) as the file
-        }
-
-        auto writer = new XMLexport(this);
-        writers.insert(filename_xml, writer);
-        writer->writeModuleXML(moduleName, filename_xml);
-
-        if (entry[1].toInt()) {
-            modulesToSync << moduleName;
-        }
 
-        if (!zipName.isEmpty()) {
             struct zip_source* s = zip_source_file(zipFile, filename_xml.toStdString().c_str(), 0, 0);
-            QTime t;
-            t.start();
-            //            int err = zip_file_add( zipFile, QString(moduleName+".xml").toStdString().c_str(), s, ZIP_FL_OVERWRITE );
-            int err = zip_add(zipFile, QString(moduleName + ".xml").toStdString().c_str(), s);
+            err = zip_add(zipFile, QString(moduleName + ".xml").toStdString().c_str(), s);
```

Comment:
:bangbang: Hang on - this is the ***other*** place where the obsolete `zip_add` should change to `zip_file_add` - and the current code ***cannot work*** because `zip_add` will not overwrite an ***existing*** file in the archive (and `zip_file_add` will only do so if given the `ZIP_FL_OVERWRITE` flag) - you could, I think, temporarily use `zip_replace` here instead of `zip_add` because this is always *updating* an existing archive isn't it...

Of course you will also want to sync your work with #1847 because you will want to use the `ZIP_FL_ENC_UTF_8` flag with the `zip_file_add`...! :wink:

Reply:
I am not sure what you are saying it should be replaced with...

Reply:
The `zip_add` here also needs to be changed, at least to `zip_replace` (which is also from the 0.1x libzip version era) if not to `zip_file_add` with the `ZIP_FL_OVERWRITE` (and the `ZIP_FL_ENC_UTF_8` for consistency with the other usage).

The bottom line is that a `moduleName.mpackage` or `moduleName.zip` (archive) opened by the old code in `void Host::saveModules(int sync, bool backup)` which is now in `void Host::updateModuleZips() const` cannot have the contained `moduleName.xml` file updated because `zip_add` will always fail to replace it with the updated version...